modules/visualization.py

```diff
@@ -0,0 +1,154 @@
+from turtle import width
+import extractor
+import crawler
+import json
+import networkx as nx
+from collections import Counter
+import matplotlib.pyplot as plt
+
+class visualize_json:
+    def __init__(self, json_file):
+        self.json_file = json_file
+        self.data = json.load(open(self.json_file))
+        self.G = nx.DiGraph()
+        self.G.add_nodes_from(self.data.keys())
+        for key,value in self.data.items():
+            self.G.add_edges_from([key,value]for value in value)
+
+    
+    def visualize(self):
+        """Visualization of the graph."""
+        
+        nx.draw(self.G, with_labels=True,node_size=150,font_size=6,node_shape="o",node_color='blue',edge_color='red',alpha=0.4,width=2)
+        plt.savefig('graph.png')
+        plt.show()
+
+
+    def indegree(self):
+        """Indegree of the graph."""
+        indegree =dict(self.G.in_degree())
+        return indegree
+
+
+    def plot_indegree(self):
+        """Plot the indegree of the graph."""
+        indegree = dict(self.G.in_degree())
+        indegree_counter = Counter(indegree.values())
+        indegree_values = list(indegree_counter.values())
+        indegree_keys = list(indegree_counter.keys())
+        plt.scatter(indegree_keys,indegree_values)
+        plt.ylabel('No. of Nodes')
+        plt.xlabel('Indegree')
+        plt.title('Indegree of the graph')
+        plt.savefig('indegree_plot.png')
```

Comment:
- Save the image using `plt.savefig(os.path.join(out_path, "visualization", "indegree_plot.png"))`.
- Do the same for all graphs.